rfcs/77-dev-tools-abstraction.md

```diff
@@ -595,101 +557,35 @@ Additional dev tool specific metadata (such as a classification scheme) can be a
 
 Optional methods on both `ModalDevTool` and `DevCommand` will allow us to override the supplied defaults if needed.
 
-We also need access to one other critical piece of information: a function pointer that allows us to construct a new value of this type from a string.
-
 As a result our `DevToolMetadata` looks like:
 
 ```rust
 struct DevToolMetaData {
     name: &'static str,
     type_id: TypeId,
     type_info: &'static TypeInfo,
-    from_str_fn: fn(&str) -> Result<Box<dyn Reflect>, DevToolParseError>,
     short_description: Option<&'static str>
 }
 ```
 
 ### What do the registries for our dev tools look like?
 
 Keeping track of the registered dev tools without storing them all in a dedicated collection is quite challenging!
-How can we keep track of it under the hood?
-
-```rust
-#[derive(Resource)]
-struct DevToolsRegistry {
-    /// The stored collection of modal dev tools, tracked in a type-erased way using [`ComponentId`]
-    /// 
-    /// The key is the `name()` provided by the `ModalDevTool` trait.
-    modal_dev_tools: HashMap<String, DevToolMetadata>,
-    /// The metadata for all registered dev commands.
-    /// 
-    /// The key is the `name()` provided by the `DevCommand` trait.
-    dev_commands: HashMap<String, DevCommandMetaData> 
-}
-```
+How can we keep track of it under the hood? We can use already existing `TypeRegistry` for that.
 
 There are a few key operations that we will need to perform:
 
 ```rust
-impl DevToolsRegistry {
-    /// Gets a reference to the specified modal dev tool by `ComponentId`.
-    fn get_tool_by_id(world: &World, component_id: ComponentId) -> Option<&dyn ModalDevTool> {
-        let resource = world.get_resource_by_id(component_id)?;
-        resource.downcast_ref()
-    }
-
-    /// Gets a mutable reference to the specified modal dev tool by `ComponentId`.
-    fn get_tool_mut_by_id(mut world: &mut World, component_id: ComponentId) -> Option<&mut dyn ModalDevTool> {
-        let resource = world.get_resource_mut_by_id(component_id)?;
-        resource.downcast_mut()
-    }
+// You can get [`ModalDevTool`] like any other resource.
+world.get_resource::<D>();
 
-    /// Gets the `DevCommandMetadata` for a given dev tool by name.
-    /// 
-    /// The supplied name should match the `DevCommand::name()` method.
-    fn get_command_metadata(name: &str) -> Option<&DevCommandMetadata> {
-        self.dev_commands.get(name)
-    }
+// todo: show how to get `Resource` from `TypeId`.
 
-    /// Gets the `DevToolMetadata` for a given dev tool by name.
-    /// 
-    /// The supplied name should match the `DevCommand::name()` method.
-    fn get_tool_metadata(name: &str) -> Option<&DevToolMetadata> {
-        self.modal_dev_tools.get(name)
-    }
+// Iterates over tools from `TypeRegistry`. Note: This is not yet implemented.
```

Comment:
`iter_with _data` needs a note in the implementation strategy section of the RFC.